applications/plugins/Communication/components/serverCommunicationOSC.inl

```diff
@@ -0,0 +1,245 @@
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2017 INRIA, USTL, UJF, CNRS, MGH                    *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include "serverCommunicationOSC.h"
+#include <Communication/components/CommunicationSubscriber.h>
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace communication
+{
+
+ServerCommunicationOSC::ServerCommunicationOSC()
+    : Inherited(), osc::OscPacketListener()
+    , d_packetSize(initData(&d_packetSize, (int)(1024), "packetSize", "OSC packet size (default=1024)"))
+{
+}
+
+ServerCommunicationOSC::~ServerCommunicationOSC()
+{
+    if (d_job.getValueString().compare("receiver") == 0)
+    {
+        m_socket->Break();
+        delete m_socket;
+    }
+    Inherited::closeCommunication();
+}
+
+ServerCommunicationOSC::OSCDataFactory* ServerCommunicationOSC::getFactoryInstance(){
+    static OSCDataFactory* s_localfactory = nullptr ;
+    if(s_localfactory==nullptr)
+        s_localfactory = new ServerCommunicationOSC::OSCDataFactory() ;
+    return s_localfactory ;
+}
+
+void ServerCommunicationOSC::initTypeFactory()
+{
+    getFactoryInstance()->registerCreator("f", new DataCreator<float>());
+    getFactoryInstance()->registerCreator("d", new DataCreator<double>());
+    getFactoryInstance()->registerCreator("i", new DataCreator<int>());
+    getFactoryInstance()->registerCreator("s", new DataCreator<std::string>());
+
+    getFactoryInstance()->registerCreator("matrixf", new DataCreator<vector<float>>());
```

Comment:
I need your help for the matrix support.
Matrix support in a nutshell, a factory is responsible to provide data depending of the type name received from the network message.
Exemple for OSC : "/aMessage f:1.0", we know the type is float (character f). If the data does not exist where we want to stock it, we ask the factory to create a data "f". In this case it will return a new DataCreator<float>().

For the matrix, a specific message is send like "/aMessage s:matrix i:1 i:3 f:1 f:2 f:3" correspond to a matrix of size 1:3 type float {1.0, 2.0, 3.0}.

For the matrix case we need a matrix wich can be resized at runtime and it have to be easily modify.

So here is the question. Which sofa type correspond to the need ?
I tried fullMatrix and didn´t succeed to read/write it. :/